core/src/io/github/lonamiwebs/klooni/game/State.java

```diff
@@ -0,0 +1,282 @@
+package io.github.lonamiwebs.klooni.game;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.math.MathUtils;
+import com.badlogic.gdx.math.Vector2;
+
+/*
+    State is an object designed to help validating randomly-generated blocks. Every time a block is
+    inserted into the board, the game's state changes. Different permutation of blocks and different
+    position creates new states. Thus to validate a set of generated blocks, we have to check all
+    possible states. Instead of checking them directly on the board, we create a special class for
+    validation.
+
+    This is used in infinite mode.
+ */
+
+public class State {
+
+    // region Members
+
+    private boolean[][] state;
+    private final int cellCount;
+    private int emptySpace = 0;
+    private static int[] unfitBlock = new int[6];
```

Comment:
I need to track and compare unfit blocks from many `State`.

Reply:
Perhaps it would be better to have a `State` per-game and reset only some fields when needed? `static` pretty much means it's a global, which for a state like this sounds bad.

Reply:
If i make `unfitBlock` as a local instance of the static method `validateBlock`, it should work, right?

Reply:
I believe yes, because `checkPermute` does `unfitBlock[i] = -1;` anyway which clears the fields. I *think* removing the `static` should work.

Reply:
Removing `static` does work. Thank you for the suggestion.